internal/webhooks/cluster.go

```diff
@@ -551,11 +551,44 @@ func ValidateClusterForClusterClass(cluster *clusterv1.Cluster, clusterClass *cl
 	return allErrs
 }
 
+// validateClusterClassExistsAndIsReconciled will try to get the ClusterClass referenced in the Cluster. If it does not exist or is not reconciled it will add a warning.
+// In any other case it will return an error.
+func (webhook *Cluster) validateClusterClassExistsAndIsReconciled(ctx context.Context, newCluster *clusterv1.Cluster) (*clusterv1.ClusterClass, admission.Warnings, error) {
+	var allWarnings admission.Warnings
+	clusterClass, clusterClassPollErr := webhook.pollClusterClassForCluster(ctx, newCluster)
+	if clusterClassPollErr != nil {
+		// Add a warning if the Class does not exist or if it has not been successfully reconciled.
+		switch {
+		case apierrors.IsNotFound(clusterClassPollErr):
+			allWarnings = append(allWarnings,
+				fmt.Sprintf(
+					"Cluster refers to ClusterClass %s in the topology but it does not exist. "+
```

Comment:
```suggestion
					"Cluster refers to ClusterClass %s in the topology but it does not exist (yet). "+
```

Reply:
"yet" might be slightly misleading here if, for example, there's a typo in `spec.topology.class`

Reply:
🤔 The suggestion here was for users that are installing CRDs all at once, and getting warnings because the ordering might be out of place, we might want to hint that things will reconcile eventually?

Reply:
Yup. I also prefer more generic warnings over misleading ones. The latter is more dangerous